src/semck/typeck.rs

```diff
@@ -1795,6 +1805,19 @@ mod tests {
     }
 
     #[test]
+    fn type_defer() {
+        ok("fun foo() { }
+            fun f() { defer foo(); }") ;
+        err("fun foo(var i: int) { return i; }
```

Comment:
`UnknownIdentifier` is already thrown in `nameck` not in `typeck`. `typeck` isn't reached anymore, therefore we do not check what we thought we would. Sorry I didn't think about that...

Use a test like: `fun foo(a: int) {} fun f() { defer foo(); }` This should give the error `ParamTypesIncompatible` and is checked in typeck.rs